utils/create-validate-install-plans.ts

```diff
@@ -1,70 +1,100 @@
-'use strict';
-const path = require('path');
-const { readYamlFile, passedProcessArguments } = require('./helpers');
-const {
+import {
+  InstallPlanDestination,
+  InstallPlanDirectiveInput,
+  InstallPlanMutationVariable,
+  InstallPlanOperatingSystem,
+  InstallPlanTargetInput,
+  InstallPlanTargetType,
+} from './types/InstallPlanMutationVariables';
+import {
+  InstallPlanConfig,
+  InstallPlanInstall,
+  InstallPlanTarget,
+} from './types/InstallPlanConfig';
+import { NerdGraphResponseWithLocalErrors } from './types/nerdgraph';
+
+import * as path from 'path';
+import { readYamlFile, passedProcessArguments } from './helpers';
+import {
   fetchPaginatedGHResults,
   filterInstallPlans,
-} = require('./github-api-helpers');
-
-const {
+} from './github-api-helpers';
+import {
   fetchNRGraphqlResults,
   translateMutationErrors,
   chunk,
-} = require('./nr-graphql-helpers');
-const { track, CUSTOM_EVENT } = require('./newrelic/customEvent');
-
-const INSTALL_PLAN_MUTATION = `# gql 
-mutation QuickstartRepoInstallPlanMutation (
-  $description: String!
-  $dryRun: Boolean
-  $displayName: String!
-  $fallback: Nr1CatalogInstallPlanDirectiveInput
-  $heading: String!
-  $id: ID!
-  $primary: Nr1CatalogInstallPlanDirectiveInput!
-  $target: Nr1CatalogInstallPlanTargetInput!
-) {
-  nr1CatalogSubmitInstallPlanStep(
-    dryRun: $dryRun
-    installPlanStep: {
-      description: $description
-      displayName: $displayName
-      fallback: $fallback
-      heading: $heading
-      id: $id
-      primary: $primary
-      target: $target
-    }
+} from './nr-graphql-helpers';
+import { track, CUSTOM_EVENT } from './newrelic/customEvent';
+
+const gql = String.raw;
+export const INSTALL_PLAN_MUTATION = gql`
+  # gql
+  mutation QuickstartRepoInstallPlanMutation(
+    $description: String!
+    $dryRun: Boolean
+    $displayName: String!
+    $fallback: Nr1CatalogInstallPlanDirectiveInput
+    $heading: String!
+    $id: ID!
+    $primary: Nr1CatalogInstallPlanDirectiveInput!
+    $target: Nr1CatalogInstallPlanTargetInput!
   ) {
-    installPlanStep {
-      id
+    nr1CatalogSubmitInstallPlanStep(
+      dryRun: $dryRun
+      installPlanStep: {
+        description: $description
+        displayName: $displayName
+        fallback: $fallback
+        heading: $heading
+        id: $id
+        primary: $primary
+        target: $target
+      }
+    ) {
+      installPlanStep {
+        id
+      }
     }
   }
-}
 `;
 
+interface InstallPlanMutationResponse {
+  installPlanStep: {
+    id: string;
+  };
+}
+
 /**
  * Builds the target parameter from the config into the variables for NR Request
  * @param {Object} target the `target` parameter object
  * @returns {Object} target transformed for NR request
```

Comment:
We probably don't want to have types defined in two places. It would be a bummer if an IDE read from these types instead of the actual Typescript types.